src/vivarium_public_health/exposure/effect.py

```diff
@@ -0,0 +1,367 @@
+"""
+###################
+# Exposure Effect #
+###################
+
+This module contains tools for modeling the relationship between risk
+exposure models and disease models.
+
+"""
+import warnings
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from importlib import import_module
+from typing import Any
+
+import numpy as np
+import pandas as pd
+import scipy
+from layered_config_tree import ConfigurationError
+from vivarium import Component
+from vivarium.framework.engine import Builder
+from vivarium.framework.values import Pipeline
+
+from vivarium_public_health.risks.data_transformations import (
+    load_exposure_data,
+    pivot_categorical,
+)
+from vivarium_public_health.utilities import EntityString, TargetString, get_lookup_columns
+
+from .exposure import Exposure
+
+
+class ExposureEffect(Component, ABC):
+    """A component to model the effect of a risk factor on an affected entity's target rate.
+
+    This component can source data either from builder.data or from parameters
+    supplied in the configuration.
+
+    For an exposure named 'exposure_name' that affects  'affected_entity' and 'affected_cause',
+    the configuration would look like:
+
+    .. code-block:: yaml
+
+       configuration:
+            exposure_effect.exposure_name_on_affected_target:
+               exposure_parameters: 2
+               incidence_rate: 10
+
+    """
+
+    def __init__(self, entity: str, target: str):
+        """
+
+        Parameters
+        ----------
+        entity
+            Type and name of exposure, supplied in the form
+            "entity.entity_name" where exposure_type should be singular (e.g.,
+            exposure instead of exposures).
+        target
+            Type, name, and target rate of entity to be affected by risk factor,
+            supplied in the form "entity_type.entity_name.measure"
+            where entity_type should be singular (e.g., cause instead of causes).
+        """
+        super().__init__()
+        self.entity = EntityString(entity)
+        self.target = TargetString(target)
+
+        self._exposure_distribution_type = None
+        self.target_pipeline_name = f"{self.target.name}.{self.target.measure}"
+        self.target_paf_pipeline_name = f"{self.target_pipeline_name}.paf"
+
+    ###############
+    # Properties #
+    ##############
+
+    @property
+    def name(self) -> str:
+        return self.get_name(self.entity, self.target)
+
+    @staticmethod
+    @abstractmethod
+    def get_name(self) -> Callable[[EntityString, TargetString], str]:
```

Comment:
Is `get_name()` actually used in a static context? Why do we need a separate method for this?